src/index.rs

```diff
@@ -278,6 +286,18 @@ impl<T: Eq + Hash + Copy + Debug> Index<T> {
         document_frequency
     }
 
+    /// Count the document frequency.
+    pub(crate) fn count_documents(&self, node_index: ArenaIndex<InvertedIndexNode<T>>) -> usize {
+        let node = self.arena_index.get(node_index).unwrap();
+        let mut pointer_option = node.first_doc;
+        let mut document_frequency = 0;
+        while let Some(pointer) = pointer_option {
+            document_frequency += 1;
+            pointer_option = self.arena_doc.get(pointer).unwrap().next;
```

Comment:
Perhaps, we should still check whether the document is removed before bumping the document_frequency? 

```suggestion
            while let Some(pointer) = pointer_option {
                let is_removed = match removed {
                    Some(set)  => {
                        if set.contains(&self.arena_doc.get(pointer).unwrap().details_key)
                        {
                            match &prev_pointer {
                                None => {
                                    node.first_doc = self.arena_doc.get(pointer).unwrap().next;
                                }
                                Some(prev) => {
                                    self.arena_doc.get_mut(*prev).unwrap().next =
                                self.arena_doc.get(pointer).unwrap().next;
                                }
                            }
                        }
                        true
                    },
                    None => false
                };
                if !is_removed
                {
                    document_frequency += 1;
                    prev_pointer = Some(pointer);
                }
                pointer_option = self.arena_doc.get(pointer).unwrap().next;
                
            }
```

I curious how this could pass without any test errors

Reply:
Edit* I updated my suggestion, I copied too freely from master

Reply:
Good catch, this should be fixed now 👍